### PR TITLE
Imgpkg upgrade to 0.29.0

### LINKED ISSUE #127: `workload create` on Windows throws "x509: certificate signed by unknown authority"
### Please fill out the issue checklist below and provide ALL the requested information.

- [x] I reviewed open and closed Github issues that may be related to my problem.
- [x] I am reporting a bug that others will be able to reproduce.

### Describe the bug
When I invoke `tanzu apps workload apply` or `tanzu apps workload create` on Windows, the command fails with exit status 1 & error:

> Writing 'index.docker.io/atwoosnam/sample-app:latest': Writing image: Retried 5 times: Get "https://index.docker.io/v2/": x509: certificate signed by unknown authority

### Expected behavior

1. No error 
2. App image gets written to the repository provided as the `--source-image` parameter

### Steps to Reproduce

1. Using a Windows OS...
3. Install tanzu cli + apps plugin
4. Clone [sample java accelerator app](https://github.com/sample-accelerators/tanzu-java-web-app) & cd into repo
5. Try to push app to TAP via `tanzu apps workload apply -f config/workload.yaml --live-update --local-path {PATH_TO_JAVA_ACCELERATOR_REPO} --source-image {YOUR_IMAGE_REPOSITORY} --namespace {YOUR_TAP_NAMESPACE} --yes`
6. Notice "`Error: Writing '{IMAGE}': Writing image: Retried 5 times: Get "https://{IMAGE_REPO_DOMAIN}/v2/": x509: certificate signed by unknown authority`"

### Screenshots
image
image

### Version (Apps plugin version, Version of K8s running on cluster)
Put the output of the following commad
```sh
tanzu version && tanzu apps version
``` 
```
version: v0.11.6
buildDate: 2022-05-18
sha: 6819c5a1
v0.5.1
```
### Environment where the bug was observed (cloud, OS, etc)
Windows 11 Enterprise, Version 21H2 (OS build 22000.675)

### Additional context & Relevant Debug Output (Logs, etc)

- I've tried changing these things without seeing the error change:
  - URL provided for `--source-image`:
    - `harbor.dhaka.cf-app.com/awoosnam`
    - `hub.docker.com/repository/docker/atwoosnam/sample-app`
    - `hub.docker.com/atwoosnam/sample-app`
    - `index.docker.io/atwoosnam/sample-app`
    - `docker.io/atwoosnam/sample-app`
    - `docker.io/rkamaldocker/sample-app`
  - different `tanzu app` commands
    - `workload create` and `workload apply` both give the same error
  - different shells in Windows:
    - Powershell (as administrator)
    - Git Bash ([info](https://appuals.com/what-is-git-bash/))
- I **am** able to push to my `atwoosnam/sample-app` image repository in DockerHub using the Docker CLI
image
- My teammate @zzori-theoriginal got the same error when invoking `tanzu plugin list` on windows:
image

### PR DESCRIPTION
**Pull request by shashwathi**
### Which issue(s) this PR fixes
Fixes #127

### CODE REVIEW

File: pkg/commands/workload_test.go [hunk 1]
```diff
@@ -1674,7 +1674,12 @@ Published source
 
 			cmd := &cobra.Command{}
 			ctx := cli.WithCommand(context.Background(), cmd)
-			ctx = source.StashGgcrRemoteOptions(ctx, remote.WithTransport(registry.Client().Transport))
+			ctx = source.StashContainerRemoteTransport(ctx, reg.Client().Transport)
+			ctx = source.StashRegistryOptions(ctx, registry.Opts{
+				VerifyCerts:           true,
+				RetryCount:            2,
```
Comment (shashwathi) [inline pkg/commands/workload_test.go:1680]: @odinnordico : I changed the defaults to what is in imgpkg CLI flags. Let me know if this is ok

**Review by odinnordico — APPROVED** [review]

File: pkg/source/imgpkg.go [hunk 2]
```diff
@@ -19,19 +19,36 @@ package source
...
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/plainimage"
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/registry"
 )
 
 func ImgpkgPush(ctx context.Context, dir string, excludedFiles []string, image string) (string, error) {
-	options := RetrieveGgcrRemoteOptions(ctx)
-	// TODO support more registry options
-	reg, err := registry.NewRegistry(registry.Opts{VerifyCerts: true}, options...)
+
+	options := RetrieveRegistryOptions(ctx)
```
Comment (scothis) [inline pkg/source/imgpkg.go:35]: **question** these default opts are so similar to the opts defined in the test case why do we need custom options for tests?

File: pkg/source/imgpkg.go [hunk 3]
```diff
@@ -19,19 +19,36 @@ package source
...
+		options = &registry.Opts{
+			VerifyCerts:           true,
+			RetryCount:            5,
+			ResponseHeaderTimeout: 30 * time.Second,
+		}
+	}
+	var transport http.RoundTripper
+	t := RetrieveStashContainerRemoteTransport(ctx)
+	if t == nil {
+		transport = remote.DefaultTransport.Clone()
```
Comment (scothis) [inline pkg/source/imgpkg.go:46]: **issue** no need to clone unless you're going to modify the transport

File: pkg/source/imgpkg.go [hunk 4]
```diff
@@ -52,16 +69,29 @@ func ImgpkgPush(ctx context.Context, dir string, excludedFiles []string, image s
...
 
-func StashGgcrRemoteOptions(ctx context.Context, options ...remote.Option) context.Context {
-	return context.WithValue(ctx, ggcrRemoteOptionsStashKey{}, options)
+func StashContainerRemoteTransport(ctx context.Context, rTripper http.RoundTripper) context.Context {
+	return context.WithValue(ctx, containerRemoteTransportStashKey{}, rTripper)
 }
 
-func RetrieveGgcrRemoteOptions(ctx context.Context) []remote.Option {
-	options, ok := ctx.Value(ggcrRemoteOptionsStashKey{}).([]remote.Option)
+func RetrieveStashContainerRemoteTransport(ctx context.Context) *http.RoundTripper {
```
Comment (scothis) [inline pkg/source/imgpkg.go:91]: the next three suggestions are part of a single change, but GitHub won't let me offer the suggestion in one go
```suggestion
func RetrieveStashContainerRemoteTransport(ctx context.Context) http.RoundTripper {
```

File: pkg/source/imgpkg.go [hunk 5]
```diff
@@ -52,16 +69,29 @@ func ImgpkgPush(ctx context.Context, dir string, excludedFiles []string, image s
...
+	transport, ok := ctx.Value(containerRemoteTransportStashKey{}).(http.RoundTripper)
 	if !ok {
-		return []remote.Option{}
+		return nil
```
Comment (scothis) [inline pkg/source/imgpkg.go:94]:
```suggestion
		return remote.DefaultTransport
```

File: pkg/source/imgpkg.go [hunk 6]
```diff
@@ -52,16 +69,29 @@ func ImgpkgPush(ctx context.Context, dir string, excludedFiles []string, image s
...
 	}
-	return options
+	return &transport
```
Comment (scothis) [inline pkg/source/imgpkg.go:96]:
```suggestion
	return transport
```

**Review by rashedkvm — APPROVED** [review]

**Review by atmandhol — APPROVED** [review]

**Pull request merged**